src/ilx/EraseUnions.fs

```diff
@@ -1082,9 +1082,8 @@ let mkClassUnionDef (addMethodGeneratedAttrs, addPropertyGeneratedAttrs, addProp
                   CustomAttrs= emptyILCustomAttrs }.WithNestedAccess(cud.cudReprAccess).WithAbstract(true).WithSealed(true).WithImport(false).WithEncoding(ILDefaultPInvokeEncoding.Ansi).WithHasSecurity(false))
 
     let baseTypeDef = 
-       { td with 
+       { td.WithInitSemantics(ILTypeInit.BeforeField) with 
           NestedTypes = mkILTypeDefs (Option.toList enumTypeDef @ altTypeDefs @ altDebugTypeDefs @ td.NestedTypes.AsList)
-          Attributes = td.Attributes ||| TypeAttributes.BeforeFieldInit
           Extends= (match td.Extends with None -> Some ilg.typ_Object | _ -> td.Extends) 
           Methods= mkILMethods (ctorMeths @ baseMethsFromAlt @ selfMeths @ tagMeths @ altUniqObjMeths @ existingMeths)
```

Comment:
@forki Note that there are already significant numbers of allocations in places like this.  It's just scratch-pad temporary allocations, it's ok generally speaking